### PR TITLE
[#10] add asynchronous tests

### PR DESCRIPTION
**Pull request by chenfanggm**
Hi Homa,

Here's a try to the issue [#10](https://github.com/unional/assert-order/issues/10).
I only did it for testing 'once' asynchronously now, and would like to hear your feedback before apply it to all other methods. Please feel free to give any feedback.

Thanks,
Chen

### CODE REVIEW

**Comment by unional** [comment]
@chenfanggm thanks! That's nicely done. I think that serves as a good starting point.
One thing I would suggest is making the test util to take a function, e.g.:

```ts
// Just ideas
let fn = createAsync(() => order.once(0))
fn = createSequential(() => order.once(0), () => order.once(1), ...)
fn = createParallel(() => order.any(1,2), () => order.some(1), ...)

// run the test function
fn()
t.throws(fn, ...)
```

We can draw some parallels from [structural programming](https://en.wikipedia.org/wiki/Structured_programming)
This would make the test code much more readable, and still have the flexibility to construct any case we want.

On the other hand, #16 would suggest to re-implement the library.

I'm happy to merge this as is, and improve upon it in subsequence PRs. 🌷

**Comment by chenfanggm** [comment]
Hi @unional ,
Thank you for your feedback. Already start to learn from it.
I have updated the PR per your suggestions, and it's truly more clean and clear now.

Thanks,
Chen

File: src/testUtils.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          resolve(fns[i]())
+        }
+        catch (err) {
+          reject(err)
+        }
+      }, 1)
+    })
+  },
+  runParallelAsync: (...fns) => {
+    const promises: Promise<any>[] = []
```
Comment (unional) [inline src/testUtils.ts]: Can simplify this by
```ts
fns.map(fn => runSync(fn))
```

You can also export each util function individually using named export. That's typically a better way to go than default export:

```ts
export function runAsync() ...
export function runParallelAsync() ...

// AssertOrder.spec.ts
import { runAsync, runParallelAsync } from './testUtil'
// or
import * as testUtil from './testUtil'
```

**Comment by unional** [comment]
Thanks! 🌷

**Pull request merged**